src/olympia/files/models.py

```diff
@@ -82,6 +83,8 @@ class File(OnChangeMixin, ModelBase):
     # STATUS_NULL means the user didn't disable the File - i.e. Mozilla did.
     original_status = models.PositiveSmallIntegerField(
         default=amo.STATUS_NULL)
+    webext_permissions_json = models.CharField(
+        max_length=1024, default='')
```

Comment:
As I mentioned before, you could use a nullable `OneToOneField` + a separate model. In any case, you want a `JSONField` (from `django_extensions.db.fields.json`) - and probably no `max_length`